src/unix/special.c

```diff
@@ -100,20 +100,24 @@ closure_function(1, 4, void, mounts_handler,
         print_uuid(b, uuid);
     push_u8(b, ' ');
     if (mount_point) {
-        while (true) {
-            int rv = file_get_path(fs, mount_point, buffer_end(b), buffer_space(b));
+        for (int count = 2; count > 0; count--) {
+            int rv = file_get_path(get_root_fs(), mount_point, buffer_end(b), buffer_space(b));
             if (rv > 0) {
                 buffer_produce(b, rv - 1);  /* drop the string terminator character */
-                break;
-            } else if (!buffer_extend(b, 2 * buffer_space(b))) {
+                goto out;
+            } else if (!buffer_extend(b, 2 * buffer_length(b))) {
```

Comment:
If file_get_path() fails on the first non-root volume due to insufficient buffer space, this code would call buffer_extend() when buffer_length() is strlen("root / tfs rw 0 0\n", i.e. 18, and since the available buffer space is (64 - 18) = 46 a buffer_extend() call with 2 * buffer_length() = 36 as argument value would be a no-op, so the second attempt at file_get_path() will also fail and this volume would be skipped. So doing just two iterations wouldn't be enough. Maybe we could iterate until the available buffer space is PATH_MAX, or alternatively call buffer_extend() with PATH_MAX as argument.

Reply:
I went with calling buffer_extend with PATH_MAX to avoid having to deal with multiple extend allocations.